### PR TITLE
libxgboost4j is now part of the CMake build

### PR DESCRIPTION
**Pull request by superbobry**
This is part of the ongoing effort to build jvm-packages on Windows. 

The `jvm` target in Makefile seems to be UNIX-focused.  Have anyone tested it MingGW/Windows?

### CODE REVIEW

**Comment by RAMitchell** [comment]
Awesome. Have you tested the GPU plugin? We don't have CI yet unfortunately. In particular I want to check cmake 3.2 is sufficient as I've had problems with older versions before.

@arnocandel is this suitable for h20.ai integration?

**Comment by superbobry** [comment]
Hey @RAMitchell, I don't have a GPU to test it, neither do Travis I suspect :)

Could you test it locally?

**Comment by RAMitchell** [comment]
I will test it in the next few days although its merged already :)

**Comment by superbobry** [comment]
Yeah, I should've added a [WIP] prefix...

File: jvm-packages/xgboost4j/src/native/xgboost4j.cpp [hunk 1]
```diff
@@ -25,7 +25,7 @@
 // set handle
 void setHandle(JNIEnv *jenv, jlongArray jhandle, void* handle) {
 #ifdef __APPLE__
-  long out = (long) handle;
+  jlong out = (long) handle;
```
Comment (superbobry) [inline jvm-packages/xgboost4j/src/native/xgboost4j.cpp]: Turns out there's a better way to do this. I'm working on a PR.

File: jvm-packages/xgboost4j/src/native/xgboost4j.cpp [hunk 2]
```diff
@@ -76,7 +76,8 @@ XGB_EXTERN_C int XGBoost4jCallbackDataIterNext(
         batch, jenv->GetFieldID(batchClass, "featureValue", "[F"));
       XGBoostBatchCSR cbatch;
       cbatch.size = jenv->GetArrayLength(joffset) - 1;
-      cbatch.offset = jenv->GetLongArrayElements(joffset, 0);
+      cbatch.offset = reinterpret_cast<long *>(
```
Comment (superbobry) [inline jvm-packages/xgboost4j/src/native/xgboost4j.cpp]: This as well.

**Pull request merged**